java/tests/com/google/privacy/differentialprivacy/CountTest.java

```diff
@@ -512,4 +513,146 @@ private static void testForBias(
     // equal to the raw count.
     assertThat(stats.mean()).isWithin(sampleTolerance).of((double) rawCount);
   }
-}
+
+  @Test
+  public void computeConfidenceInterval_negativeBounds() {
+    when(noise.computeConfidenceInterval(
+            anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(-5,-3));
+    count.computeResult();
+
+    // The result interval = (-5, -3), but count can't be negative, then it should be clamped to (0, 0).
+    assertThat(count.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(0, 0));
+  }
+
+  @Test
+  public void computeConfidenceInterval_positiveBounds() {
+    when(noise.computeConfidenceInterval(
+            anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(5,3));
+    count.computeResult();
+
+    // The result interval is positive, then it should not be clamped.
+    assertThat(count.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(5, 3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_negativeLowerBound() {
+    when(noise.computeConfidenceInterval(
+            anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenReturn(ConfidenceInterval.create(-5,8));
+    count.computeResult();
+
+    // The result interval = (-5, 8), but count can't be equal to -5 then it should be clamped to (0, 8).
+    assertThat(count.computeConfidenceInterval(0.152145599))
+            .isEqualTo(ConfidenceInterval.create(0, 8));
+  }
+
+  @Test
+  public void computeConfidenceInterval_infiniteBounds() {
```

Comment:
it test checks 2 things at once 

1. clamps negative inf to 0 (it can be additional test )
2. rounds double inf to long.max . So I would add this as infiniteUpperBound_roundsToMaxLong